### ISSUE TITLE
Register an existing component (/catalog-import) , Link to a repository always creates a PR although a catalog-info.yaml file already present in the root of the provided repo

### ISSUE THREAD

**Issue by sgantaya** [backstage/backstage#22162]
### 📜 Description

As per the documentation when we input a GitHub repo URL ex: https://github.com/backstage/backstage, while Register an existing component (/catalog-import), the wizard should discovers all catalog-info.yaml files in the repository, previews the entities, and adds them to the Scaffolded Backstage App catalog.

But although there is a catalog-info.yaml file exist in the root of the provided repo, the wizard is always trying to create a PR adding a new catalog-info.yaml.

### 👍 Expected behavior

The wizard should discovers all catalog-info.yaml files in the repository, previews the entities, and adds them to the Scaffolded Backstage App catalog

### 👎 Actual Behavior with Screenshots

![image](https://github.com/backstage/backstage/assets/66821171/ac06c913-90a8-4fbc-a5c1-a7cb6f1b27d0)
![image](https://github.com/backstage/backstage/assets/66821171/46ad24ba-9d15-4ded-a584-186cb9b864fc)
![image](https://github.com/backstage/backstage/assets/66821171/2ff81060-6c54-495b-8fa6-56fa3cec997d)

### 👟 Reproduction steps

1. Go to http://localhost:3000/catalog-import
2. Enter 'https://github.com/backstage/backstage' in URL input
3. click on ANALYZE button

### 🖥️ Your Environment

OS:   Windows_NT 10.0.22621 - win32/x64
node: v18.18.0
yarn: 1.22.19
cli:  0.25.0 (installed)
backstage:  1.21.0

Dependencies:
  @backstage/app-defaults                                          1.4.6
  @backstage/backend-app-api                                       0.5.9
  @backstage/backend-common                                        0.20.0
  @backstage/backend-dev-utils                                     0.1.2
  @backstage/backend-openapi-utils                                 0.1.1
  @backstage/backend-plugin-api                                    0.6.8
  @backstage/backend-tasks                                         0.5.13
  @backstage/catalog-client                                        1.5.1
  @backstage/catalog-model                                         1.4.3
  @backstage/cli-common                                            0.1.13
  @backstage/cli-node                                              0.2.1
  @backstage/cli                                                   0.25.0
  @backstage/config-loader                                         1.6.0
  @backstage/config                                                1.1.1
  @backstage/core-app-api                                          1.11.2
  @backstage/core-compat-api                                       0.1.0
  @backstage/core-components                                       0.13.9
  @backstage/core-plugin-api                                       1.8.1
  @backstage/e2e-test-utils                                        0.1.0
  @backstage/errors                                                1.2.3
  @backstage/eslint-plugin                                         0.1.4
  @backstage/frontend-plugin-api                                   0.4.0
  @backstage/integration-aws-node                                  0.1.8
  @backstage/integration-react                                     1.1.22
  @backstage/integration                                           1.8.0
  @backstage/plugin-api-docs                                       0.10.2
  @backstage/plugin-app-backend                                    0.3.56
  @backstage/plugin-app-node                                       0.1.8
  @backstage/plugin-auth-backend-module-atlassian-provider         0.1.0
  @backstage/plugin-auth-backend-module-gcp-iap-provider           0.2.2
  @backstage/plugin-auth-backend-module-github-provider            0.1.5
  @backstage/plugin-auth-backend-module-gitlab-provider            0.1.5
  @backstage/plugin-auth-backend-module-google-provider            0.1.5
  @backstage/plugin-auth-backend-module-oauth2-provider            0.1.5
  @backstage/plugin-auth-backend-module-oauth2-proxy-provider      0.1.0
  @backstage/plugin-auth-backend-module-okta-provider              0.0.1
  @backstage/plugin-auth-backend                                   0.20.2
  @backstage/plugin-auth-node                                      0.4.2
  @backstage/plugin-catalog-backend-module-scaffolder-entity-model 0.1.5
  @backstage/plugin-catalog-backend                                1.16.0
  @backstage/plugin-catalog-common                                 1.0.19
  @backstage/plugin-catalog-graph                                  0.3.2
  @backstage/plugin-catalog-import                                 0.10.4
  @backstage/plugin-catalog-node                                   1.6.0
  @backstage/plugin-catalog-react                                  1.9.2
  @backstage/plugin-catalog                                        1.16.0
  @backstage/plugin-events-node                                    0.2.17
  @backstage/plugin-github-actions                                 0.6.9
  @backstage/plugin-org                                            0.6.18
  @backstage/plugin-permission-common                              0.7.11
  @backstage/plugin-permission-node                                0.7.19
  @backstage/plugin-permission-react                               0.4.18
  @backstage/plugin-proxy-backend                                  0.4.6
  @backstage/plugin-scaffolder-backend-module-azure                0.1.0
  @backstage/plugin-scaffolder-backend-module-bitbucket            0.1.0
  @backstage/plugin-scaffolder-backend-module-gerrit               0.1.0
  @backstage/plugin-scaffolder-backend-module-github               0.1.0
  @backstage/plugin-scaffolder-backend-module-gitlab               0.2.11
  @backstage/plugin-scaffolder-backend                             1.19.2
  @backstage/plugin-scaffolder-common                              1.4.4
  @backstage/plugin-scaffolder-node                                0.2.9
  @backstage/plugin-scaffolder-react                               1.7.0
  @backstage/plugin-scaffolder                                     1.17.0
  @backstage/plugin-search-backend-module-catalog                  0.1.12
  @backstage/plugin-search-backend-module-pg                       0.5.17
  @backstage/plugin-search-backend-module-techdocs                 0.1.12
  @backstage/plugin-search-backend-node                            1.2.12
  @backstage/plugin-search-backend                                 1.4.8
  @backstage/plugin-search-common                                  1.2.9
  @backstage/plugin-search-react                                   1.7.4
  @backstage/plugin-search                                         1.4.4
  @backstage/plugin-tech-radar                                     0.6.11
  @backstage/plugin-techdocs-backend                               1.9.1
  @backstage/plugin-techdocs-module-addons-contrib                 1.1.3
  @backstage/plugin-techdocs-node                                  1.11.0
  @backstage/plugin-techdocs-react                                 1.1.14
  @backstage/plugin-techdocs                                       1.9.2
  @backstage/plugin-user-settings                                  0.7.14
  @backstage/release-manifests                                     0.0.11
  @backstage/test-utils                                            1.4.6
  @backstage/theme                                                 0.5.0
  @backstage/types                                                 1.1.1
  @backstage/version-bridge                                        1.0.7

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by jithenms** [comment]
I'd be happy to take a look at this. I also received the same behavior when testing this locally.

**Comment by jithenms** [comment]
@sgantaya it looks like this has to do with the GitHub code search being moved to a backend plugin in a previous release which means you must add it to the CatalogBuilder in order to enable it. Further details below:

https://backstage.io/docs/releases/v1.7.0-changelog/#backstageplugin-catalog-import090

**Comment by tudi2d** [comment]
Thanks for raising this issue! The behaviour is expected, but our documentation might not reflect this good enough. Would you be interested in improving the documentation to make this clear? :)

**Comment by mcaxtrzen** [comment]
@jithenms would you mind clarifying where is this catalog.ts file that the link you posted mentioned? I'm using the new backend module

**Comment by briemarie** [comment]
@mcaxtrzen did you ever figure this out? I can't find it either.

**Comment by mcaxtrzen** [comment]
@briemarie I have not unfortunately

**Comment by briemarie** [comment]
@jithenms or @tudi2d can you clarify what this catalog.ts file is supposed to be or point us to the documentation for this new process? I can't find anything in the docs and this is blocking us. thank you!

**Comment by awanlin** [comment]
Hi @briemarie, the `catalog.ts` file referenced in the linked changelog is from the Legacy Backend, you won't have that anymore. It's also terribly old so I would not trust that being correct nearly 20 versions later.

This issue is still easy to reproduce with the latest version - `1.39.0` - released today so I'll reopen this, contributions welcome!

**Comment by Jelledb** [comment]
Yeah this also doesn't work for me on a fresh instance of Backstage. I ended up importing all files manually by referencing the direct branch + file name.

When I enter the repo and click Analyze it just continues and says no catalog-info.yaml is found
